layers/core/gradient.py

```diff
@@ -45,6 +51,39 @@ def maxValue(self):
     def maxValue(self, maxValue):
         typeChecker(type(self).__name__, maxValue, int, "maxValue")
         self.__maxValue = maxValue
+        self._compute_curve()
+
+    def _compute_curve(self):
+        """
+            Computes the gradient color curve
+        """
+        if self.maxValue is not None and self.minValue is not None and self.colors is not None:
+            if len(self.colors) == 2:
+                s_c = colour.Color(self.colors[0])
+                e_c = colour.Color(self.colors[1])
+                self.curve = list(s_c.range_to(e_c, self.maxValue - self.minValue))
+            else:
+                s_c = colour.Color(self.colors[0])
+                m_c = colour.Color(self.colors[1])
+                e_c = colour.Color(self.colors[2])
```

Comment:
This doesn't work for gradients that have more than three colors.

Example gradient with 10 colors:
![image](https://user-images.githubusercontent.com/28868888/84040245-60948980-a970-11ea-8d9f-1c269472d99a.png)

This loop should print colors spanning the entire gradient, but as you can see it only spans the first three colors:
![image](https://user-images.githubusercontent.com/28868888/84040279-6d18e200-a970-11ea-82a9-f0b3baf58932.png)

Reply:
I think I got it sorted out.